addons/web_editor/static/src/components/html_history_dialog/html_history_dialog.js

```diff
@@ -0,0 +1,115 @@
+/** @odoo-module **/
+
+import { Dialog } from '@web/core/dialog/dialog';
+import { formatDateTime } from '@web/core/l10n/dates';
+import { useService } from "@web/core/utils/hooks";
+import { memoize } from "@web/core/utils/functions";
+import {Component, onMounted, useState, markup} from '@odoo/owl';
+import { _t } from "@web/core/l10n/translation";
+const { DateTime } = luxon;
+
+class HtmlHistoryDialog extends Component {
+    static template = "web_editor.HtmlHistoryDialog";
+    static components = { Dialog };
+    static props = {
+        close: Function,
+        restoreRequested: Function,
+        recordIds: Array,
+        versionnedFieldName: String,
+    };
+
+    state = useState({
+        revisionsData: [],
+        revisionContent: null,
+        revisionComparison: null,
+        revisionId: null,
+        asyncPending: true,
+    });
+    setup() {
+        this.size = 'xl';
+        this.title = _t("History");
+        this.orm = useService("orm");
+
+        onMounted(() => this.init());
+    }
+
+    async init() {
+        await this.getRecordData();
+        this.state.asyncPending = false;
+        await this.updateCurrentRevision(
+            this.state.revisionsData[0].id);
+    }
+
+    async getRecordData(limit=100, offset=0) {
+        // limit the number of record and order them by id desc
+        this.state.revisionsData = await this.orm.call(
+            "field.html.history.revision",
+            "search_read",
+            [],
+            {
+                fields: ["id", "create_date", "create_uid"],
+                domain: [
+                    ["id", "in", this.props.recordIds],
+                    ["res_field", "=", this.props.versionnedFieldName]
+                ],
+                offset: offset,
+                limit: limit,
+                order: "id desc"
+            }
+        );
+    }
```

Comment:
I don't see the point of this function. It's only used once and it is literally just one instruction/ It even has side effects on `this` even though the function is called "getSomething". I would inline it at call site.

Reply:
I've put it in a separate function to future prof it when we will nee to add pagination of the history